src/components/logging.ts

```diff
@@ -12,221 +12,372 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-import * as util from "util";
-import winston, { Logger, format, Logform } from "winston";
-import chalk from "chalk";
-import * as Transport from 'winston-transport';
-
-type LogLevel = "debug"|"info"|"warn"|"error";
-
-const CHALK_LEVELS: Record<LogLevel, string> = {
-    "debug": "blue",
-    "info": "green",
-    "warn": "yellow",
-    "error": "red",
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { ILogger, LogLevel as BotSdkLogLevel, LogService } from "matrix-bot-sdk";
+import util from "util";
+import winston, { format } from "winston";
+
+type MsgType = string|Error|any|{error?: string};
+
+/**
+ * Tries to filter out noise from the bot-sdk.
+ * @param messageOrObjects A list of values being logged.
+ * @returns True is the message is noise, or false otherwise.
+ */
+function isMessageNoise(messageOrObjects: MsgType[]) {
+	return !!messageOrObjects.find(messageOrObject => {
```

Comment:
```suggestion
	return messageOrObjects.some(messageOrObject => {
```

There's ~an app~ a method for that. :green_apple:

Reply:
fancy!